src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Download/PackageDownloadRunner.cs

```diff
@@ -48,28 +50,54 @@ public static async Task<int> RunAsync(PackageDownloadArgs args, CancellationTok
 
         public static async Task<int> RunAsync(PackageDownloadArgs args, ILoggerWithColor logger, IReadOnlyList<PackageSource> packageSources, ISettings settings, CancellationToken token)
         {
-            // Check for insecure sources
-            if (DetectAndReportInsecureSources(args.AllowInsecureConnections, packageSources, logger))
+            var packageSourceMapping = PackageSourceMapping.GetPackageSourceMapping(settings);
+            var hasSourcesArg = args.Sources != null && args.Sources.Count > 0;
+            var mappingDisabled = (packageSourceMapping != null && !packageSourceMapping.IsEnabled) || packageSourceMapping == null;
+            bool ignorePackageSourceMapping = hasSourcesArg || mappingDisabled;
+
+            // Validate all configured sources only when mapping is disabled
```

Comment:
can you help me understand this. 

My instinct is always validate the sources to be used regardless of how they're configured.